frontend/sozisel-app/src/components/RegisterScreen/RegisterScreen.tsx

```diff
@@ -0,0 +1,141 @@
+import * as yup from "yup";
+
+import { ReactElement, useState } from "react";
+
+import Button from "../utils/Button/Button";
+import Card from "../utils/Card/Card";
+import ErrorMessage from "../utils/Input/ErrorMessage";
+import Input from "../utils/Input/Input";
+import Navbar from "../Navbar/Navbar";
+import Spinner from "../utils/Spinner/Spinner";
+import conference_img from "../../assets/conference_img.png";
+import { useForm } from "react-hook-form";
+import { useHistory } from "react-router";
+import { useRegisterMutation } from "../../graphql";
+import { useTranslation } from "react-i18next";
+import { yupResolver } from "@hookform/resolvers/yup";
+
+interface RegisterFormData {
+  email: string;
+  firstName: string;
+  lastName: string;
+  password: string;
+  confirmPassword: string;
+}
+
+const registerSchema = (invalidEmailFormat: string, fieldRequired: string) => {
+  return yup.object().shape({
+    email: yup.string().email(invalidEmailFormat).required(fieldRequired),
+    firstName: yup.string().required(fieldRequired),
+    lastName: yup.string().required(fieldRequired),
+    password: yup.string().required(fieldRequired),
+    confirmPassword: yup.string().required(fieldRequired),
+  });
+};
+
+export default function Register(): ReactElement {
+  const { t } = useTranslation("common");
+
+  const [registerMutation, { loading }] = useRegisterMutation();
+  const [error, setError] = useState<string | null>(null);
+  const history = useHistory();
+  const { register, handleSubmit, errors } = useForm({
+    resolver: yupResolver(
+      registerSchema(
+        t("errorMessages.invalidEmailFormat"),
+        t("errorMessages.fieldRequired")
+      )
+    ),
+  });
+
+  const onSubmit = async (registerFormData: RegisterFormData) => {
+    setError(null);
+    if (registerFormData.password !== registerFormData.confirmPassword) {
+      setError(t("errorMessages.passwordsNotMatch"));
```

Comment:
I believe that `useForm` hook provides `setErrors` or something like that, then you can display the error under the field it belongs.